utils/src/main/java/uk/gov/pay/commons/utils/xray/XRaySessionProfiler.java

```diff
@@ -79,6 +59,51 @@ public Object profileExecutionOfQuery(DatabaseQuery databaseQuery, Record record
         }
     }
 
+    private String getDatabaseHostnameAndMetadata(DatabaseQuery databaseQuery, AbstractSession abstractSession, HashMap<String, Object> databaseMetadata) {
+        String hostname = "database";
+
+        Accessor accessor = abstractSession.getAccessor();
+        if (accessor == null) {
+            return hostname;
+        }
+        Connection connection = accessor.getConnection();
+        if (connection == null) {
+            return hostname;
+        }
+        updateMetadata(databaseQuery, databaseMetadata, connection);
+
+        return getDatabaseHostName(connection);
+    }
+
+    private void updateMetadata(DatabaseQuery databaseQuery, HashMap<String, Object> databaseMetadata, Connection connection) {
+        try {
+            DatabaseMetaData metadata = connection.getMetaData();
+            databaseMetadata.put("url", metadata.getURL());
+            databaseMetadata.put("user", metadata.getUserName());
+            databaseMetadata.put("driver_version", metadata.getDriverVersion());
+            databaseMetadata.put("database_type", metadata.getDatabaseProductName());
+            databaseMetadata.put("database_version", metadata.getDatabaseProductVersion());
+            databaseMetadata.put("preparation", databaseQuery.isCallQuery() ? "call" : "statement");
+            databaseMetadata.put("sanitized_query", StringUtils.isEmpty(databaseQuery.getSQLString()) ? "" : databaseQuery.getSQLString());
+        } catch (SQLException exception) {
+            logger.warn("Error getting database connection details.");
+        }
+    }
```

Comment:
I think this method would be better if it were `private void getMetadata(DatabaseQuery databaseQuery, Connection connection)` and returned a map (rather than mutating one):

```java
Map<String, Object> databaseMetadata = new HashMap<>();
try {
    // …
} catch (SQLException exception) {
    logger.warn("Error getting database connection details.");
}
return databaseMetadata;
```

Reply:
I'd also like to see the exception being passed to the logger